# Making Data Management Decisions

##### Preview

In this assignment you will find three data managed variables, provided from NESARC codebook, as frequency distributions. First one is the variable [S1Q2K]BIOLOGICAL OR ADOPTIVE PARENT DIE BEFORE RESPONDENT WAS 18 and i've deleted the row unknow,and set to nan the row BL.Then my second variable is [CONSUMER] indicates wheter a person is actually a aalchol drinker or not.the third variable is [S4AQ20B] and is talking about the necessity of drink alchool to feel better.
The last variable ['S2AQ8D'] is related about the quantity of alcohol the past in the past 12 months.I'll start separete the youngests of 18 and then divide it in two subsets.
   

In [66]:
import pandas 
import numpy as np
import seaborn
import matplotlib.pyplot as plt
df = pandas.read_csv('nesarc.csv')

df.head()

Unnamed: 0  ETHRACE2A ETOTLCA2  IDNUM    PSU  STRATUM      WEIGHT  CDAY  \
0           0          5               1   4007      403 3928.613505    14   
1           1          5   0.0014      2   6045      604 3638.691845    12   
2           2          5               3  12042     1218 5779.032025    23   
3           3          5               4  17099     1704 1071.754303     9   
4           4          2               5  17099     1704 4986.952377    18   

   CMON  CYEAR  ...  SOLP12ABDEP  HAL12ABDEP  HALP12ABDEP  MAR12ABDEP  \
0     8   2001  ...     0.000000    0.000000     0.000000    0.000000   
1     1   2002  ...     0.000000    0.000000     0.000000    0.000000   
2    11   2001  ...     0.000000    0.000000     0.000000    0.000000   
3     9   2001  ...     0.000000    0.000000     0.000000    0.000000   
4    10   2001  ...     0.000000    0.000000     0.000000    0.000000   

   MARP12ABDEP  HER12ABDEP  HERP12ABDEP  OTHB12ABDEP  OTHBP12ABDEP  NDSymptoms  
0     0.000000    0.000000     0.000000     0.000000      0.000000         nan  
1     0.000000    0.000000     0.000000     0.000000      0.000000         nan  
2     0.000000    0.000000     0.000000     0.000000      0.000000         nan  
3     0.000000    0.000000     0.000000     0.000000      0.000000         nan  
4     0.000000    0.000000     0.000000     0.000000      0.000000         nan  

[5 rows x 3010 columns]

In [67]:
pandas.set_option('display.float_format' , lambda x:'%f'%x)
df.describe()

Unnamed: 0   ETHRACE2A       IDNUM          PSU     STRATUM  \
count 8185.000000 8185.000000 8185.000000  8185.000000 8185.000000   
mean  4092.000000    2.040073 4093.000000 29636.264997 2958.929016   
std   2362.950310    1.538717 2362.950310 15772.518085 1568.717690   
min      0.000000    1.000000    1.000000  1001.000000  101.000000   
25%   2046.000000    1.000000 2047.000000 13097.000000 1308.000000   
50%   4092.000000    1.000000 4093.000000 31098.000000 3109.000000   
75%   6138.000000    2.000000 6139.000000 42095.000000 4216.000000   
max   8184.000000    5.000000 8185.000000 56017.000000 5605.000000   

            WEIGHT        CDAY        CMON       CYEAR      REGION  ...  \
count  8185.000000 8185.000000 8185.000000 8185.000000 8185.000000  ...   
mean   4942.638069   16.225779    8.713256 2001.129872    2.639829  ...   
std    3347.099407    8.738690    2.993762    0.336183    0.891212  ...   
min     398.855701    1.000000    1.000000 2001.000000    1.000000  ...   
25%    2717.532245    9.000000    8.000000 2001.000000    2.000000  ...   
50%    3827.362169   16.000000    9.000000 2001.000000    3.000000  ...   
75%    6986.299036   24.000000   11.000000 2001.000000    3.000000  ...   
max   37648.529080   31.000000   12.000000 2002.000000    4.000000  ...   

       SOLP12ABDEP  HAL12ABDEP  HALP12ABDEP  MAR12ABDEP  MARP12ABDEP  \
count  8184.000000 8184.000000  8184.000000 8184.000000  8184.000000   
mean      0.004765    0.000611     0.017351    0.015152     0.084800   
std       0.088998    0.024711     0.164538    0.169147     0.357382   
min       0.000000    0.000000     0.000000    0.000000     0.000000   
25%       0.000000    0.000000     0.000000    0.000000     0.000000   
50%       0.000000    0.000000     0.000000    0.000000     0.000000   
75%       0.000000    0.000000     0.000000    0.000000     0.000000   
max       3.000000    1.000000     3.000000    3.000000     3.000000   

       HER12ABDEP  HERP12ABDEP  OTHB12ABDEP  OTHBP12ABDEP  NDSymptoms  
count 8184.000000  8184.000000  8184.000000   8184.000000  261.000000  
mean     0.000489     0.005254     0.000000      0.000489    2.421456  
std      0.034954     0.115294     0.000000      0.027074    1.786315  
min      0.000000     0.000000     0.000000      0.000000    0.000000  
25%      0.000000     0.000000     0.000000      0.000000    1.000000  
50%      0.000000     0.000000     0.000000      0.000000    2.000000  
75%      0.000000     0.000000     0.000000      0.000000    4.000000  
max      3.000000     3.000000     0.000000      2.000000    6.000000  

[8 rows x 682 columns]

In [68]:
df = df.replace(r'^\s*$', np.NaN, regex=True)

In [69]:
df1=df[(df['AGE']==18)]
df1

Unnamed: 0  ETHRACE2A ETOTLCA2  IDNUM    PSU  STRATUM      WEIGHT  CDAY  \
3              3          5      NaN      4  17099     1704 1071.754303     9   
9              9          5   0.1565     10   4013      404 3008.554718    16   
36            36          5      NaN     37  36094     3616 2292.549993    17   
153          153          4      NaN    154   6098      624 6853.268126    28   
156          156          5      NaN    157  36094     3616 1586.297176    29   
...          ...        ...      ...    ...    ...      ...         ...   ...   
7914        7914          1   0.6544   7915  39091     3913 7850.316750    30   
7950        7950          3   0.0025   7951  34011     3407 9030.603521    28   
8032        8032          1      NaN   8033  13097     1306 8657.814391    17   
8086        8086          1   0.0744   8087  46001     4601 3791.054404    30   
8138        8138          1      NaN   8139  31032     3106 6809.443772    26   

      CMON  CYEAR  ...  SOLP12ABDEP  HAL12ABDEP  HALP12ABDEP  MAR12ABDEP  \
3        9   2001  ...     0.000000    0.000000     0.000000    0.000000   
9        8   2001  ...     0.000000    0.000000     0.000000    1.000000   
36       1   2002  ...     0.000000    0.000000     0.000000    0.000000   
153      9   2001  ...     0.000000    0.000000     0.000000    0.000000   
156      9   2001  ...     0.000000    0.000000     0.000000    0.000000   
...    ...    ...  ...          ...         ...          ...         ...   
7914    12   2001  ...     0.000000    0.000000     0.000000    0.000000   
7950     8   2001  ...     0.000000    0.000000     0.000000    0.000000   
8032    10   2001  ...     0.000000    0.000000     0.000000    0.000000   
8086     8   2001  ...     0.000000    0.000000     0.000000    0.000000   
8138     9   2001  ...     0.000000    0.000000     0.000000    0.000000   

      MARP12ABDEP  HER12ABDEP  HERP12ABDEP  OTHB12ABDEP  OTHBP12ABDEP  \
3        0.000000    0.000000     0.000000     0.000000      0.000000   
9        0.000000    0.000000     0.000000     0.000000      0.000000   
36       0.000000    0.000000     0.000000     0.000000      0.000000   
153      0.000000    0.000000     0.000000     0.000000      0.000000   
156      0.000000    0.000000     0.000000     0.000000      0.000000   
...           ...         ...          ...          ...           ...   
7914     0.000000    0.000000     0.000000     0.000000      0.000000   
7950     0.000000    0.000000     0.000000     0.000000      0.000000   
8032     0.000000    0.000000     0.000000     0.000000      0.000000   
8086     0.000000    0.000000     0.000000     0.000000      0.000000   
8138     0.000000    0.000000     0.000000     0.000000      0.000000   

      NDSymptoms  
3            nan  
9            nan  
36           nan  
153          nan  
156          nan  
...          ...  
7914    1.000000  
7950         nan  
8032         nan  
8086    0.000000  
8138    4.000000  

[128 rows x 3010 columns]

In [70]:
#DRANK ALCOHOL TO IMPROVE MOOD IN LAST 12 MONTHS S4A
#CONSUMER DRINKING STATUS consumer
df1=df1[['S1Q2K','S4AQ20B','CONSUMER','S2AQ8D']]  
 
df1

S1Q2K S4AQ20B  CONSUMER S2AQ8D
3        2     NaN         2    NaN
9        2     NaN         1     10
36       2     NaN         3    NaN
153      2     NaN         3    NaN
156      2     NaN         3    NaN
...    ...     ...       ...    ...
7914     2     NaN         1      9
7950     2     NaN         1     10
8032     2     NaN         3    NaN
8086     2     NaN         1      9
8138     2     NaN         3    NaN

[128 rows x 4 columns]

In [71]:
#unique values
df1['S1Q2K'].unique()

array(['2', nan, '1', '9'], dtype=object)

In [72]:
#recode some columns with different name variables
df1=df1.rename(columns={'S4AQ20B': 'ALC4MOOD', 'S2AQ8D': 'ALCL_12MO'})

In [73]:
#remap the indicators of ALCL_12MO QUANTITY
df1["ALCL_12MO"].replace({"10": "4", "9": "4",'8':'4','7':'4','6':'3','5':'3','4':'2','3':'2','2':'1','1':'1'}, inplace=True)

In [74]:
df1['S1Q2K'] = pandas.to_numeric(df1['S1Q2K'], errors='coerce')
df1 = df1.dropna(subset=['S1Q2K'])
df1['S1Q2K'] = df1['S1Q2K'].astype(int)

In [75]:
df1['S1Q2K'].replace([9], np.nan, inplace=True)
df1['S1Q2K'].replace([0], np.nan, inplace=True)
df1['ALCL_12MO'] = df1['ALCL_12MO'].replace(['99'],np.nan)

In [76]:
df1= df1.fillna(0)

In [77]:
#dead parewnts before 18
subset1 = df1[(df1['S1Q2K'] ==1)]
subset1

S1Q2K ALC4MOOD  CONSUMER ALCL_12MO
945  1.000000        2         1         4
3540 1.000000        0         3         0
4747 1.000000        0         1         4
4867 1.000000        0         2         0
5629 1.000000        0         2         0
6560 1.000000        2         1         4
7193 1.000000        0         3         0

In [78]:
#no dead parents before 18
subset2 = df1[(df1['S1Q2K'] ==2)]
subset2

S1Q2K ALC4MOOD  CONSUMER ALCL_12MO
3    2.000000        0         2         0
9    2.000000        0         1         4
36   2.000000        0         3         0
153  2.000000        0         3         0
156  2.000000        0         3         0
...       ...      ...       ...       ...
7914 2.000000        0         1         4
7950 2.000000        0         1         4
8032 2.000000        0         3         0
8086 2.000000        0         1         4
8138 2.000000        0         3         0

[116 rows x 4 columns]

In [79]:
# dead before 18 values
df1['S1Q2K'].value_counts(dropna=True)

2.000000    116
1.000000      7
0.000000      1
Name: S1Q2K, dtype: int64

In [80]:
# alchool to improve mood users
df1['ALC4MOOD'].value_counts(dropna=True)#102 Nan values deleted

0    102
2     19
1      3
Name: ALC4MOOD, dtype: int64

In [81]:
# total alohol drinkers under 18 years old
df1['CONSUMER'].value_counts(dropna=True)

1    59
3    55
2    10
Name: CONSUMER, dtype: int64

In [82]:
#quantity of alchol in a timeframe
df1['ALCL_12MO'].value_counts(dropna=True) #69 Nan values deleted

0    66
4    48
3     8
2     2
Name: ALCL_12MO, dtype: int64

In [83]:
deads = df1.groupby('S1Q2K').size() 
deads

S1Q2K
0.000000      1
1.000000      7
2.000000    116
dtype: int64

In [84]:
deads_prc = df1.groupby('S1Q2K').size() * 100 / len(df1)   #percentages of parents died before respondent has 18
deads_prc

S1Q2K
0.000000    0.806452
1.000000    5.645161
2.000000   93.548387
dtype: float64

In [85]:
impr_mood= df1.groupby('ALC4MOOD').size() 
impr_mood

ALC4MOOD
0    102
1      3
2     19
dtype: int64

In [86]:
impr_mood_per=df1.groupby('ALC4MOOD').size() * 100 / len(df1)
impr_mood_per

ALC4MOOD
0   82.258065
1    2.419355
2   15.322581
dtype: float64

In [87]:
consumr= df1.groupby('CONSUMER').size() 
consumr

CONSUMER
1    59
2    10
3    55
dtype: int64

In [88]:
consumr_per=df1.groupby('CONSUMER').size() * 100 / len(df1)
consumr_per


CONSUMER
1   47.580645
2    8.064516
3   44.354839
dtype: float64

In [89]:
often_drink=df1.groupby('ALCL_12MO').size() 
often_drink

ALCL_12MO
0    66
2     2
3     8
4    48
dtype: int64

In [90]:
often_drink=df1.groupby('ALCL_12MO').size() * 100 / len(df1)
often_drink

ALCL_12MO
0   53.225806
2    1.612903
3    6.451613
4   38.709677
dtype: float64

In [91]:
#relation between  childresns with dead parents before 18 and if they are current alcohol drinkers
relation1= subset1.groupby('CONSUMER').size()
relation1

CONSUMER
1    3
2    2
3    2
dtype: int64

In [92]:
#percentage of alchool consumer with dad parents
relation1p=subset1.groupby('CONSUMER').size() * 100 / len(subset1)
relation1p

CONSUMER
1   42.857143
2   28.571429
3   28.571429
dtype: float64

In [93]:
#relation between  childresns with no dead parents before 18 and if they are current alcohol drinkers
relation2=subset2.groupby('CONSUMER').size()
relation2

CONSUMER
1    55
2     8
3    53
dtype: int64

In [94]:
#percentage of alchool consumer with no dead parents
relation2p=subset2.groupby('CONSUMER').size() * 100 / len(subset2)
relation2p

CONSUMER
1   47.413793
2    6.896552
3   45.689655
dtype: float64

Firstly, a new 2 variables called ‘subset1’ and 'subset2'  was necessary to be created in order to divide the partecipants, then i dropped Na and unkown information so this could cause some different resulte in this project, In addition i changed the variable name['S4AQ20B'] in ['ALC4MOOD'] and ['S2AQ8D'] in ['ALCL_12MO'] finally i I have related the subset 1(parents died before child has 18) with 'current drinkers' variable and we can ntice that (in a really small quantity)  the death of parents before 18 can affect the child and he could be a drinker.On the other side children in the subset two we can notice a high percentage of lifetime absatiner and current drinkers.

In [95]:
subset2.dtypes

S1Q2K        float64
ALC4MOOD      object
CONSUMER       int64
ALCL_12MO     object
dtype: object

In [124]:
#def a functioninorder to 
def ALCOHOLDIST (row):
    if row['ALC4MOOD'] ==1 :
        return 5
    if row['ALCL_12MO'] == 1 :
        return 1
    if row['ALCL_12MO'] == 2 :
        return 2
    if row['ALCL_12MO'] == 3 :
        return 3
    if row['ALCL_12MO'] == 4 :
        return 4

In [125]:
subset1['ALC4MOOD'] = subset1['ALC4MOOD'].astype(float)
subset1['ALCL_12MO'] = subset1['ALCL_12MO'].astype(float)

In [126]:
subset1['ALCOHOLDIST'] = subset1.apply (lambda row: ALCOHOLDIST (row), axis=1)
subsetfin1 =subset1[['S1Q2K','ALCL_12MO', 'ALC4MOOD','ALCOHOLDIST']].copy()
subsetfin1.style.set_precision(0)


In [127]:
subset2['ALC4MOOD'] = subset2['ALC4MOOD'].astype(float)
subset2['ALCL_12MO'] = subset2['ALCL_12MO'].astype(float)

In [128]:
subset2['ALCOHOLDIST'] = subset2.apply (lambda row: ALCOHOLDIST (row), axis=1)
subsetfin2 =subset2[['S1Q2K','ALCL_12MO', 'ALC4MOOD','ALCOHOLDIST']].copy()
subsetfin2.style.set_precision(0)


In [129]:
finl_prcntg1=subsetfin1.groupby('ALCOHOLDIST').size() * 100 / len(subset1)
finl_prcntg1

ALCOHOLDIST
4.000000   42.857143
dtype: float64

In [130]:
finl_prcntg2=subsetfin2.groupby('ALCOHOLDIST').size() * 100 / len(subset2)
finl_prcntg2

ALCOHOLDIST
2.000000    1.724138
3.000000    6.896552
4.000000   35.344828
5.000000    2.586207
dtype: float64

##### CONCLUSION

Finally with all of those observation we can conclude that the new variable ['ALCOHOLDIST'] related with both of the subsets, childrens whose have experienced the death of a family member(parents) ['S1Q2K'] and quantity of alcohol. the 42.8% drank only between 1-6 times during the past 12 months. Now with the other subset n.2 childrens with no death of any family member just 1.7 % drank between 2 and 4 times a week, and 6.8% use to drink between 1 and 3 times a month and the bigger part 35.3% of the partecipants use to drink just between 1 to 11 times a year.No correlation with subset1 and the variable ['ALC4MOOD'] alchool to improve mood, and just a small percentage of 2.5 in the subset2.

##### LIMITATIONS

In [ ]:
we need to take intoaccoutn that this project has a lot of miss values so this could impact the final result.